test/functional/legacy/019_smarttab_expandtab_spec.lua

```diff
@@ -0,0 +1,56 @@
+-- Tests for "r<Tab>" with 'smarttab' and 'expandtab' set/not set.
+-- Also test that dv_ works correctly
+
+local helpers = require('test.functional.helpers')
+local feed, insert = helpers.feed, helpers.insert
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+describe('performing "r<Tab>" with \'smarttab\' and \'expandtab\' set/not set, and "dv_"', function()
+  setup(clear)
+
+  it('is working', function()
+    insert([[
+      start text
+      		some test text
+      test text
+      		other test text
+          a cde
+          f ghi
+      test text
+        Second line beginning with whitespace]])
+
+    execute('set smarttab expandtab ts=8 sw=4')
+    -- Make sure that backspace works, no matter what termcap is used.
+    execute('set t_kD=x7f t_kb=x08')
+
+    execute('/some')
+    feed('r	')
+    execute('set noexpandtab')
+    execute('/other')
+    feed('r	<cr>')
+    -- Test replacing with Tabs and then backspacing to undo it.
+    feed('0wR			<esc><cr>')
```

Comment:
The `^H^H^H` is present in the file, but I think GitHub ignores them when displaying. But it's a good idea anyway to change to `<bs>` anyway, thanks!